local/kaltura/auth.php

```diff
@@ -120,13 +119,18 @@
 
 if ($ok) {
 	$course = $DB->get_record('course', array('id' => $courseid), '*', MUST_EXIST);
-	if ($id) {
-
+	if ($id && $course) {
+        $module["course"]=$course;
 		$editor = $SESSION->editor;
 		list($endpoint, $params) = local_kaltura_lti1p3_get_launch_data($module, null, $editor, $nonce);
-	}
+	} else {
+        $ok = false;
+        $error = 'course_not_found';
+        $desc = 'Course Not Found';
```

Comment:
Please add this string to the translation files, assuming this string is going to be in the UI.